mas-foundation/src/androidTest/java/com/ca/mas/foundation/PerformanceComparisonStorageTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2017 CA. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ *
+ */
+package com.ca.mas.foundation;
+
+import com.ca.mas.MASLoginTestBase;
+import com.ca.mas.core.storage.MASSecureSharedStorage;
+import com.ca.mas.core.storage.Storage;
+import com.ca.mas.core.storage.StorageException;
+import com.ca.mas.core.storage.implementation.MASStorageManager;
+
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.fail;
+
+public class PerformanceComparisonStorageTest extends MASLoginTestBase {
+
+    private final String accountName = "testPerformanceEncrypted";
+    String keyName = "key";
+    String value = "test value";
+
+    @Test
+    public void testCompareEncryptedACVSOld() throws StorageException {
+
+        // - Encrypted, shared, AccountManager
```

Comment:
@spetrov In this tests, you can see all the combinations for the MASSecureSharedStorage, it can help for better understanding of facing the automatization :)